tests/test_pygtracker.py

```diff
@@ -26,6 +28,71 @@ def test_version():
 
 # Start tests for rank_students
 
+# tests that the user inputs the right number of students
+def test_rank_students_num_of_students_match():
+    tracker = generate_input_calculate_final_grade()
+    assert (tracker.rank_students(n=4).shape[0]) == 4, "Number of students don't match"
+
+
+# raises an error when a course id is not a part of the dataset
+def test_rank_students_course_input_not_exist():
+    tracker = generate_input_calculate_final_grade()
+    with raises(NameError):
+        tracker.rank_students(course_id="500", n=4)
+
+
+def test_rank_students_ascending_input_not_bool():
+    gradetracker = pygtracker.GradeTracker()
+    with raises(TypeError):
+        gradetracker.rank_students(course_id="511", ascending="True")
+
+
+def test_rank_students_n_input_not_integer():
+    gradetracker = pygtracker.GradeTracker()
+    with raises(TypeError):
+        gradetracker.rank_students(n="4")
+
+
+def test_rank_students_course_input_not_integer():
```

Comment:
I think we are testing that it fails because the input is an integer, and not a string?
```suggestion
def test_rank_students_course_input_not_string():
```

Reply:
Agreed - updated.